File: TeamCode/src/main/java/org/riverdell/robotics/autonomous/movement/purepursuit/PurePursuitPositionChangeAction.java
package org.riverdell.robotics.autonomous.movement.purepursuit;

import org.jetbrains.annotations.NotNull;
import org.riverdell.robotics.autonomous.movement.PositionChangeAction;
import org.riverdell.robotics.autonomous.movement.PositionChangeAction;

import io.liftgate.robotics.mono.pipeline.RootExecutionGroup;

public class PurePursuitPositionChangeAction extends PositionChangeAction {
    public PurePursuitPositionChangeAction(@NotNull RootExecutionGroup executionGroup,
                                           @NotNull PurePursuitPath purePursuitPath) {
        super(null, executionGroup);
        withCustomPathAlgorithm(new PathAlgorithm(
                purePursuitPath::calculateTargetPose,
                (robotPose, targetPose) -> purePursuitPath.isFinished()
        ));
    }
}

File: TeamCode/src/main/java/org/riverdell/robotics/autonomous/movement/purepursuit/PurePursuitPositionChangeAction.kt
package org.riverdell.robotics.autonomous.movement.purepursuit

import io.liftgate.robotics.mono.pipeline.RootExecutionGroup
import org.riverdell.robotics.autonomous.movement.PositionChangeAction

class PurePursuitPositionChangeAction(
    executionGroup: RootExecutionGroup,
    purePursuitPath: PurePursuitPath
) : PositionChangeAction(null, executionGroup)
{
    init
    {
        withCustomPathAlgorithm(PathAlgorithm(
            { currentPose -> purePursuitPath.calculateTargetPose(currentPose) },
            { _, _ -> purePursuitPath.isFinished }
        ))
    }
}
